code/game/objects/items/weapons/tanks/watertank.dm

```diff
@@ -288,42 +287,40 @@
 	var/Adj = user.Adjacent(target)
 	if(Adj)
 		AttemptRefill(target, user)
-	if(nozzle_mode == NANOFROST)
-		if(Adj)
-			return //Safety check so you don't blast yourself trying to refill your tank
-		var/datum/reagents/R = reagents
-		if(R.total_volume < 100)
-			to_chat(user, "You need at least 100 units of water to use the nanofrost launcher!")
-			return
-		if(nanofrost_cooldown)
-			to_chat(user, "Nanofrost launcher is still recharging")
-			return
-		nanofrost_cooldown = 1
-		R.remove_any(100)
-		var/obj/effect/nanofrost_container/A = new /obj/effect/nanofrost_container(get_turf(src))
-		log_game("[key_name(user)] used Nanofrost at [get_area(user)] ([user.x], [user.y], [user.z]).")
-		playsound(src,'sound/items/syringeproj.ogg',40,1)
-		for(var/a=0, a<5, a++)
-			step_towards(A, target)
-			sleep(2)
-		A.Smoke()
-		spawn(100)
-			if(src)
-				nanofrost_cooldown = 0
-		return
-	if(nozzle_mode == METAL_FOAM)
-		if(!Adj|| !isturf(target))
-			return
-		if(metal_synthesis_cooldown < 5)
-			var/obj/effect/particle_effect/foam/F = new /obj/effect/particle_effect/foam(get_turf(target), 1)
+
+	switch(nozzle_mode)
+		if(NANOFROST)
+			if(Adj)
+				return //Safety check so you don't blast yourself trying to refill your tank
+			if(reagents.total_volume < 100)
+				to_chat(user, "<span class='notice'>You need at least 100 units of water to use the nanofrost launcher!</span>")
+				return
+			if(nanofrost_cooldown)
+				to_chat(user, "<span class='notice'>Nanofrost launcher is still recharging.</span>")
+				return
+			nanofrost_cooldown = TRUE
+			reagents.remove_any(100)
+			var/obj/effect/nanofrost_container/A = new /obj/effect/nanofrost_container(get_turf(src))
+			log_game("[key_name(user)] used Nanofrost at [get_area(user)] ([user.x], [user.y], [user.z]).")
+			playsound(src,'sound/items/syringeproj.ogg', 40, TRUE)
+			for(var/a in 1 to 6)
+				step_towards(A, target)
+				sleep(2)
+			A.Smoke()
+			addtimer(VARSET_CALLBACK(src, nanofrost_cooldown, FALSE))
+		if(METAL_FOAM)
+			if(!Adj)
+				return
+			if(metal_synthesis_cooldown >= 5)
+				to_chat(user, "<span class='notice'>Metal foam mix is still being synthesized.</span>")
+				return
+			var/obj/effect/particle_effect/foam/F = new /obj/effect/particle_effect/foam(get_turf(target), TRUE)
 			F.amount = 0
 			metal_synthesis_cooldown++
-			spawn(100)
-				if(src)
-					metal_synthesis_cooldown--
-		else
-			to_chat(user, "Metal foam mix is still being synthesized.")
-			return
+			addtimer(CALLBACK(src, PROC_REF(cool_down)), 10 SECONDS)
+
+/obj/item/extinguisher/mini/nozzle/proc/cool_down()
+	metal_synthesis_cooldown--
```

Comment:
cool down or cooldown here? And could maybe add "synthesis" to the name too.

Reply:
Renamed it to `metal_cooldown`